test_opensearchpy/test_krbconnection.py

```diff
@@ -0,0 +1,19 @@
+""" Test module for the RequestsKrbConnections """
+
+import unittest
+from unittest.mock import patch
+
+from opensearchpy.connection.krb_requests import RequestsKrbConnection
+
+class TestKrbConnection(unittest.TestCase): # pylint: disable=invalid-name
+    """ Test for the RequestsKrbConnection """
+
+    @patch('kerberos.authGSSClientStep')
+    @patch('kerberos.authGSSClientResponse')
+    def test_krbConnection(self, authGSSClientResport, authGSSClientStep): # pylint: disable=unused-argument,invalid-name
+        """ Check if we can we a Kerberos connection, mocking up the kerberos calls """
+        my_object = RequestsKrbConnection()
+        self.assertTrue('Authorization' in my_object.session.headers)
```

Comment:
Good start. Mock a little further, ensure that the `kerberos.` methods are called, and add a check for the value, it should be "Negotiate [dummy token returned from a mock]" or something like that.